pkg/client/command_line.go

```diff
@@ -15,62 +16,215 @@ func AddArmadaApiConnectionCommandlineArgs(rootCmd *cobra.Command) {
-		os.Exit(1)
-	} else {
-		exeDir := filepath.Dir(exePath)
-		viper.SetConfigFile(exeDir + "/armadactl-defaults.yaml")
-		err := viper.ReadInConfig()
-		if err != nil {
-			switch err.(type) {
-			case viper.ConfigFileNotFoundError:
-			case *os.PathError:
-				// No default config is fine
-			default:
-				log.Errorf("Can't read config file %s because %s\n", viper.ConfigFileUsed(), err)
-				os.Exit(1)
-			}
+	exeDir := filepath.Dir(exePath)
+	configPath := filepath.Join(exeDir, "/armadactl-defaults.yaml")
+	viper.SetConfigFile(configPath)
+	if err := viper.ReadInConfig(); err != nil {
+		if _, ok := err.(viper.ConfigFileNotFoundError); ok {
+			// Config file not found; ignore
+		} else {
+			return fmt.Errorf("[LoadCommandlineArgsFromConfigFile] error reading config file %s: %s", viper.ConfigFileUsed(), err)
 		}
 	}
 
-	if cfgFile != "" {
-		// Use config file from the flag.
+	// if no cfgFile is provided, use $HOME/.armadactl
+	if len(cfgFile) > 0 {
 		viper.SetConfigFile(cfgFile)
 	} else {
-		// Find home directory.
-		home, err := homedir.Dir()
+		homeDir, err := os.UserHomeDir()
 		if err != nil {
-			log.Error(err)
-			os.Exit(1)
+			return fmt.Errorf("[LoadCommandlineArgsFromConfigFile] error getting user home directory: %s", err)
 		}
-
-		viper.AddConfigPath(home)
+		viper.AddConfigPath(homeDir)
 		viper.SetConfigName(".armadactl")
 	}
 
-	viper.AutomaticEnv() // read in environment variables that match
+	// read environment variables
+	viper.AutomaticEnv()
 
-	// If a config file is found, read it in.
-	err = viper.MergeInConfig()
-
-	if err != nil {
-		switch err.(type) {
-		case viper.ConfigFileNotFoundError:
-			// This only occurs when looking for the default .armadactl file and it is not present
-			// This is not an error as users don't have to specify it, so do nothing
-		default:
-			log.Errorf("Can't read config file %s because %s\n", viper.ConfigFileUsed(), err)
-			os.Exit(1)
+	// merge in new config with those loaded from armadactl-defaults.yaml
+	// (note the call to viper.MergeInConfig instead of viper.ReadInConfig)
+	if err := viper.MergeInConfig(); err != nil {
+		if _, ok := err.(viper.ConfigFileNotFoundError); ok {
+			// Config file not found; ignore
+		} else {
+			return fmt.Errorf("[LoadCommandlineArgsFromConfigFile] error reading config file %s: %s", viper.ConfigFileUsed(), err)
 		}
 	}
+
+	return nil
 }
 
+// ExtractCommandlineArmadaApiConnectionDetails extracts Armada server connection details from the
+// config loaded into viper. Hence, this function must be called after loading config into viper,
+// e.g., by calling LoadCommandlineArgsFromConfigFile.
 func ExtractCommandlineArmadaApiConnectionDetails() *ApiConnectionDetails {
 	apiConnectionDetails := &ApiConnectionDetails{}
 	viper.Unmarshal(apiConnectionDetails)
 	return apiConnectionDetails
 }
+
+// ConfigYamlString returns a string representation of the merged configuration stored in viper.
+func ConfigYamlString() (string, error) {
```

Comment:
Is this used?

Reply:
It will be used by the `armadactl view` command when added. For now I'm using it for debugging.